osmoutils/partialord/partialord_test.go

```diff
@@ -0,0 +1,30 @@
+package partialord_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/osmosis-labs/osmosis/v7/osmoutils/partialord"
+)
+
+func TestAPI(t *testing.T) {
+	// begin block use case, we have a dozen modules, but only care about a couple orders.
+	// In practice this will be gotten from some API, e.g. app.AllModuleNames()
+	moduleNames := []string{"auth", "authz", "bank", "capabilities",
+		"staking", "distribution", "epochs", "mint", "upgrades", "wasm", "ibc",
+		"ibctransfers", "bech32ibc"}
+	beginBlockOrd := partialord.NewPartialOrdering(moduleNames)
+	beginBlockOrd.FirstElements("upgrades", "epochs", "capabilities")
+	beginBlockOrd.After("ibctransfers", "ibc")
+	beginBlockOrd.After("bech32ibc", "ibctransfers")
+	beginBlockOrd.Before("mint", "distribution")
+	// This is purely just to test last functionality, doesn't make sense in context
+	beginBlockOrd.LastElements("auth", "authz", "wasm")
```

Comment:
This should be what the app.go logic ends up looking like. 

I think in practice, we will just use `Before` syntax, and not mix `After` and `Before`

Reply:
I don't think so, what makes sense depends on the context of a given application

Reply:
If my understanding of the inverse relationship between `Before` and `After` is right (i.e. `beginBlockOrd.After("ibctransfers", "ibc")` is the same as `beginBlockOrd.Before("ibc", "ibctransfers")`)

I'm still not quite sure, why do we need to expose both? In what context do we need both if we can substitute one for the other by switching the order of arguments?

Reply:
Similar reason to why int libraries expose `>` and `<=`